torch/distributed/optim/named_optimizer.py

```diff
@@ -171,9 +190,7 @@ def load_state_dict(self, state_dict: Mapping[str, Any]) -> None:
             )
         for idx, param_key in enumerate(self.ordered_param_keys):
             if param_key not in state.keys():
-                raise ValueError(
```

Comment:
Can we add a comment to describe why this is legit?

Reply:
Sure. Let me explain it with comments.